### PR TITLE
chore: use 'rank' terminology consistently for sorted sets, rather than 'index'

### PR DESCRIPTION
**Pull request by cprice404**

### CODE REVIEW

File: src/internal/cache-client.ts [hunk 1]
```diff
@@ -1797,34 +1799,34 @@ export class CacheClient {
...
-    startIndex: number,
-    endIndex?: number
+    startRank: number,
+    endRank?: number
   ): Promise<CacheSortedSetFetch.Response> {
     try {
       validateCacheName(cacheName);
       validateSortedSetName(sortedSetName);
-      validateSortedSetIndices(startIndex, endIndex);
+      validateSortedSetIndices(startRank, endRank);
```
Comment (malandis) [inline src/internal/cache-client.ts:1812]: validateSortedSetRanks?

File: src/simple-cache-client.ts [hunk 2]
```diff
@@ -917,11 +917,11 @@ export class SimpleCacheClient {
    *
    * @param {string} cacheName - The cache containing the sorted set.
    * @param {string} sortedSetName - The sorted set to fetch from.
-   * @param {SortedSetFetchByIndexOptions} options
-   * @param {number} [options.startIndex] - The index of the first element to
+   * @param {SortedSetFetchByRankOptions} options
+   * @param {number} [options.startRank] - The index of the first element to
```
Comment (nand4011) [inline src/simple-cache-client.ts:921]: The description still uses index

File: src/simple-cache-client.ts [hunk 3]
```diff
@@ -917,11 +917,11 @@ export class SimpleCacheClient {
...
    * fetch. Defaults to 0. This index is inclusive, ie the element at this index
    * will be fetched.
-   * @param {number} [options.endIndex] - The index of the last element to fetch.
+   * @param {number} [options.endRank] - The index of the last element to fetch.
```
Comment (nand4011) [inline src/simple-cache-client.ts:924]: Same here

File: test/integration/sorted-set.test.ts [hunk 4]
```diff
@@ -281,11 +281,11 @@ describe('Integration tests for sorted set operations', () => {
       });
 
       it('should fetch only the specified range if start index is specified', async () => {
```
Comment (nand4011) [inline test/integration/sorted-set.test.ts:283]: A bunch of test descriptions still say index

**Pull request merged**